docs/04-api-reference/front-commerce-core/testing.mdx

```diff
@@ -0,0 +1,58 @@
+---
+title: "Testing"
+sidebar_position: 9
+description:
+  "Testing is an important part of the development process. It helps to ensure that
+the code is working as expected and that it is not breaking any existing
+functionality. In this section, we will cover all the tools and utilities that
+we provide for testing."
+---
+
+<p>{frontMatter.description}</p>
+
+<SinceVersion tag="3.4" />
+
+## [`createFrontCommerceProviderStub`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/core/react/FrontCommerceProviderStub.tsx?ref_type=heads)
+
+This function is used to create a stub for the
+[`FrontCommerceProvider`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/core/react/components.tsx?ref_type=heads#L29-69)
+component. It is useful for testing components that depend on the
+`FrontCommerceProvider` component.
+
+### Props (optional)
+
+- `device` : Device type
+  [`Device`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/core/react/MediaQuery.tsx?ref_type=heads#L8-12)
+- `messages`: Translations messages `Record<string, string>`
+- `isContributionModeEnabled`: Contribution mode status `boolean`
+- `publicConfig`: Public config
+  [`PublicConfig`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/core/config/publicConfigProvider.ts#L28-90)
```

Comment:
**nit(non-blocking):** this could be formatted in a table like some other examples in the reference api: https://github.com/front-commerce/developers.front-commerce.com/blob/fc-2034-vite-routes-from-extensions/docs/03-extensions/magento1/reference/axios-instances.mdx#L22